File: java/calendar/src/test/java/com/rbleggi/calendar/CalendarConflictTest.java
package com.rbleggi.calendar;

import org.junit.jupiter.api.Test;
import java.time.LocalDateTime;
import java.util.Set;
import static org.junit.jupiter.api.Assertions.*;

class CalendarConflictTest {
    @Test
    void bookMeetingRejectOverlapping() {
        var calendar = new Calendar();
        var user = new User("1", "Alice");
        var m1 = new Meeting("m1", "Sync", LocalDateTime.of(2025, 1, 1, 9, 0), LocalDateTime.of(2025, 1, 1, 10, 0), Set.of(user));
        var m2 = new Meeting("m2", "Conflict", LocalDateTime.of(2025, 1, 1, 9, 30), LocalDateTime.of(2025, 1, 1, 10, 30), Set.of(user));
        calendar.bookMeeting(m1);
        assertFalse(calendar.bookMeeting(m2));
    }

    @Test
    void conflictDoesNotAddToList() {
        var calendar = new Calendar();
        var user = new User("1", "Alice");
        var m1 = new Meeting("m1", "Sync", LocalDateTime.of(2025, 1, 1, 9, 0), LocalDateTime.of(2025, 1, 1, 10, 0), Set.of(user));
        var m2 = new Meeting("m2", "Conflict", LocalDateTime.of(2025, 1, 1, 9, 30), LocalDateTime.of(2025, 1, 1, 10, 30), Set.of(user));
        calendar.bookMeeting(m1);
        calendar.bookMeeting(m2);
        assertEquals(1, calendar.getAllMeetings().size());
    }
}

File: kotlin/calendar/src/test/kotlin/com/rbleggi/calendar/CalendarConflictTest.kt
package com.rbleggi.calendar

import java.time.LocalDateTime
import kotlin.test.Test
import kotlin.test.assertFalse
import kotlin.test.assertEquals

class CalendarConflictTest {
    @Test
    fun `bookMeeting rejects overlapping meeting`() {
        val calendar = Calendar()
        val user = User("1", "Alice")
        val meeting1 = Meeting(title = "Test1", start = LocalDateTime.of(2025, 1, 1, 9, 0), end = LocalDateTime.of(2025, 1, 1, 10, 0), attendees = setOf(user))
        val meeting2 = Meeting(title = "Test2", start = LocalDateTime.of(2025, 1, 1, 9, 30), end = LocalDateTime.of(2025, 1, 1, 10, 30), attendees = setOf(user))
        calendar.bookMeeting(meeting1)
        assertFalse(calendar.bookMeeting(meeting2))
    }

    @Test
    fun `conflict does not add to list`() {
        val calendar = Calendar()
        val user = User("1", "Alice")
        val meeting1 = Meeting(title = "Test1", start = LocalDateTime.of(2025, 1, 1, 9, 0), end = LocalDateTime.of(2025, 1, 1, 10, 0), attendees = setOf(user))
        val meeting2 = Meeting(title = "Test2", start = LocalDateTime.of(2025, 1, 1, 9, 30), end = LocalDateTime.of(2025, 1, 1, 10, 30), attendees = setOf(user))
        calendar.bookMeeting(meeting1)
        calendar.bookMeeting(meeting2)
        assertEquals(1, calendar.getAllMeetings().size)
    }
}
